Unit type in tuple struct not parsed correctly

The unit type `()` is not parsed correctly and emit an error.

I tried this code:

```rust
pub struct Foo(pub ());
```

I expected to see this happen: 

No error

Instead, this happened:
```
./test.rs:1:21: error: unexpected token ‘)’ in visibility
    1 | pub struct Foo(pub ());
      |                     ^
./test.rs:1:22: error: unrecognised token ‘)’ in type
    1 | pub struct Foo(pub ());
      |                      ^
./test.rs:1:22: error: could not parse type in tuple struct field
```

### Meta

007166c

--- a/gcc/rust/parse/rust-parse-impl.h
+++ b/gcc/rust/parse/rust-parse-impl.h
@@ -2296,8 +2296,11 @@ Parser<ManagedTokenSource>::parse_visibility ()
   auto vis_loc = lexer.peek_token ()->get_locus ();
   lexer.skip_token ();
 
-  // create simple pub visibility if no parentheses
-  if (lexer.peek_token ()->get_id () != LEFT_PAREN)
+  // create simple pub visibility if
+  // - found no parentheses
+  // - found unit type `()`
+  if (lexer.peek_token ()->get_id () != LEFT_PAREN
+      || lexer.peek_token (1)->get_id () == RIGHT_PAREN)
     {
       return AST::Visibility::create_public (vis_loc);
       // or whatever

--- a/gcc/testsuite/rust/compile/parse_pub_unit_type.rs
+++ b/gcc/testsuite/rust/compile/parse_pub_unit_type.rs
@@ -1,0 +1,1 @@
+pub struct Foo(pub ());

